### DISCUSSION TITLE
Gocryptfs for 6 To of data

### DISCUSSION THREAD

**Discussion by ADMECA** [rfjakob/gocryptfs#860] [Q&A]
hello, I have about 6TB of data, (about 100000 files and the largest file is 120GB) to copy to a CIFS share. is gocryptfs robust enough to handle the encryption of such a large amount of data ?

Thanks for your help.

**Comment by celesteking** [comment]
@rfjakob where are you, dear? 
I've got the same question, running gocrypt over cifs/smb3. Right now I'm getting very bad performance from it.

**Comment by rfjakob** [comment]
Tge volume of data is no problem, but there's this problem with cifs:
https://github.com/rfjakob/gocryptfs/issues/532

It seems to be a bug with the linux kernel and/or the smb server.

This https://github.com/restic/restic/issues/2659#issuecomment-1546701199
suggests that the problem has been solved in latest linux kernels.

So before deleting the original files, do make sure that all files in
gocryptfs are readable. "gocryptfs -fsck" checks that.

Or get the sha256sum of all original files and check the copy.

**Comment by ADMECA** [comment]
Very interesting, thanks you for the links, i will read all of this.
